src/vs/platform/webContentExtractor/common/webContentExtractor.ts

```diff
@@ -3,16 +3,28 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { VSBuffer } from '../../../base/common/buffer.js';
+import { CancellationToken } from '../../../base/common/cancellation.js';
 import { URI } from '../../../base/common/uri.js';
 import { createDecorator } from '../../instantiation/common/instantiation.js';
 
 export const IWebContentExtractorService = createDecorator<IWebContentExtractorService>('IWebContentExtractorService');
+export const ISharedWebContentExtractorService = createDecorator<ISharedWebContentExtractorService>('ISharedWebContentExtractorService');
+
 
 export interface IWebContentExtractorService {
 	_serviceBrand: undefined;
 	extract(uri: URI[]): Promise<string[]>;
 }
 
+/*
+ * A service that extracts image content from a given arbitrary URI. This is done in the shared process to avoid running non trusted application code in the main process.
+ */
+export interface ISharedWebContentExtractorService {
+	_serviceBrand: undefined;
+	readImage(uri: URI, token?: CancellationToken): Promise<VSBuffer | undefined>;
```

Comment:
It's generally better to require the `token` and let callers pass in `CancellationToken.None` if they need to. Otherwise it's easy to overlook the token arg